AxonDeepSeg/train_network.py

```diff
@@ -75,59 +73,88 @@ def train_model(path_trainingset, path_model, config, path_model_init=None,
     no_valid_images = int(len(os.listdir(path_validation_set)) / 2)
     valid_ids = [str(i) for i in range(no_valid_images)]
 
-    # Loading the Training images and masks in batch
-    train_gen = DataGen(train_ids, path_training_set, batch_size=no_train_images, image_size=image_size,
-                        thresh_indices=thresh_indices)
-    image_train_gen, mask_train_gen = train_gen.__getitem__(0)
 
-    # Loading the Validation images and masks in batch
-    valid_gen = DataGen(valid_ids, path_validation_set, batch_size=no_valid_images, image_size=image_size,
-                        thresh_indices=thresh_indices)
-    image_valid_gen, mask_valid_gen = valid_gen.__getitem__(0)
 
     ###################################################################################################################
     ############################################# DATA AUGMENTATION ##################################################
     ###################################################################################################################
 
-    # Data dictionary to feed into image generator
-    data_gen_args = dict(horizontal_flip=True,
-                         # flipping()[1],
-                         vertical_flip=True,
-                         # preprocessing_function = elastic
-                         # flipping()[0],
-                         # rotation_range = random_rotation()[0],
-                         # width_shift_range = shifting(patch_size, n_classes)[1],
-                         # height_shift_range = shifting(patch_size, n_classes) [0],
-                         # fill_mode = "constant",
-                         # cval = 0
-                         )
+    shifting = config["da-0-shifting-activate"],
+    rescaling = config["da-1-rescaling-activate"]
+    rotation = config["da-2-random_rotation-activate"]
+    elastic = config["da-3-elastic-activate"]
+    flipping = config["da-4-flipping-activate"]
+    gaussian_blur = config["da-5-gaussian_blur-activate"]
+
+    p_shift = p_rescale = p_rotate = p_elastic = p_flip = p_blur = 0
+    if shifting:
+        p_shift = 0.5
```

Comment:
Could you write in a comment what this/these values represent, and why you chose them to be 0.5 if the value is on? It begins with p, so I assume it has something to do with probability, but it's not clear to me why 0.5 values are chosen for all of these settings.

Reply:
Done in commit 0c6a981.